pkg/controller/storagecluster/reconcile.go

```diff
@@ -371,10 +371,13 @@ func (r *ReconcileStorageCluster) reconcileNodeTopologyMap(sc *ocsv1.StorageClus
 						updated = true
 					}
 				}
-				if strings.Contains(label, "rack") {
-					if !nodeRacks.Contains(value, node.Name) {
-						nodeRacks.Add(value, node.Name)
-					}
+			}
+			if strings.Contains(label, "rack") {
+				if !nodeRacks.Contains(value, node.Name) {
+					nodeRacks.Add(value, node.Name)
+				}
+				if !topologyMap.Contains(label, value) {
+					topologyMap.Add(label, value)
```

Comment:
I'm confused by the code:

A little above, we have the hunk:
```
if strings.Contains(label, key) {
   if !topologyMap.Contains(label, value) {
      topologyMap.Add(label, value)
   }
} 
```

Now you add:
```
if strings.Contains(label, "rack") {
   if !topologyMap.Contains(label, value) {
      topologyMap.Add(label, value)
   }
}
```
This looks pretty much like a special case. What is really changed in the resulting topology? :thinking:

Reply:
So I think, being a special case, this does not change the resulting topology map at all, and can be omitted. (making the patch even simpler)

Reply:
Just confirmed locally that the test passes without this hunk.